src/Sifo/Bootstrap.php

```diff
@@ -110,25 +110,29 @@ public static function execute( $instance_name, $controller_name = null, $contai
      * @return Controller
      * @throws Exception_DependencyInjector
      */
-    public static function invokeController( $controller )
+    public static function invokeController( $controller, $container = null )
     {
+        static::setContainer($container);
+
         $class = self::convertToControllerClassName( $controller );
 
-        if (self::$container->has($class)) {
-            $controller = self::$container->get($class);
+        if (static::$container->has($class)) {
+            $controller = static::$container->get($class);
         } else {
             /** @var Controller $controller */
             $controller = new $class();
         }
 
-        $controller->setContainer(self::$container);
+        $controller->setContainer(static::$container);
 
         return $controller;
     }
 
 
-    private static function convertToControllerClassName( $controller ): string
+    private static function convertToControllerClassName( $controller, $container = null ): string
     {
+        static::setContainer($container);
+
```

Comment:
```suggestion
    private static function convertToControllerClassName( $controller ): string
    {
```

Because we call the `static::setContainer` in the `invokeController` method so we don't need to do it again.

Also, checking the base branch, we're using return types and I'm not sure if we have to support previous PHP versions that don't have this typehint.